addons/project/models/project.py

```diff
@@ -1333,6 +1344,16 @@ def _compute_dependent_tasks_count(self):
             for task in tasks_with_dependency:
                 task.dependent_tasks_count = dependent_tasks_count_dict.get(task.id, 0)
 
+    @api.depends('depend_on_ids', 'depend_on_ids.stage_id', 'depend_on_ids.stage_id.fold')
```

Comment:
When we put `depend_on_ids.is_closed`, the `depend_on_ids` field is implicitly added. 🙂 
Moreover, `is_closed` is a related field for `stage_id.fold`.
```suggestion
    @api.depends('depend_on_ids.is_closed')
```